MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift

```diff
@@ -160,39 +119,91 @@ private extension FarmerDetailsView {
                 }
                 Spacer()
                 phoneButton
-                    .frame(width: 80, height: 35)
-                    .background(.ultraThinMaterial)
-                    .clipShape(.capsule)
-                    .padding()
-                    .shadow(radius: 4)
             }
             Divider()
-            VStack(alignment: .leading) {
-                Text("Email")
-                    .foregroundStyle(.secondary)
-                    .font(.subheadline)
-                Link(viewModel.email!, destination: URL(string: viewModel.email!)!)
+            HStack {
+                VStack(alignment: .leading) {
+                    Text("Email")
+                        .foregroundStyle(.secondary)
+                        .font(.subheadline)
+                    Text(viewModel.email ?? "Aucune adresse email")
+                        .font(.callout)
+                }
+                Spacer()
+                mailButton
             }
             Divider()
-            VStack(alignment: .leading) {
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Site Web")
+                    .foregroundStyle(.secondary)
+                    .font(.subheadline)
                 if !viewModel.websites.isEmpty {
-                    Text("Site Web")
-                        .foregroundStyle(.secondary)
-                        .font(.subheadline)
                     ForEach(viewModel.websites, id: \.id) { website in
-                        Link(website.websiteType.name, destination: URL(string: website.url)!)
+                        Text(.init("[\(website.websiteType.name)](\(website.url))"))
                     }
+                } else {
+                    Text("Aucun site web")
+                        .font(.callout)
                 }
             }
         }
         .padding()
         .background(.ultraThinMaterial)
         .clipShape(.rect(cornerRadius: 20))
+        .shadow(radius: 8)
+    }
+
+    private var directionButton: some View {
+        Button {
+            viewModel.onItineraryButtonTapped()
+        } label: {
+            Image(systemName: "map.fill")
+        }
+        .frame(width: 80, height: 40)
+        .background(.ultraThinMaterial)
+        .clipShape(.capsule)
+        .padding()
+        .shadow(radius: 4)
+    }
+
+    private var phoneButton: some View {
+        VStack {
+            if viewModel.phoneNumber == nil {
+                Image(systemName: "phone.fill")
+            } else {
+                Link(destination: viewModel.getPhoneCallURL) {
+                    Image(systemName: "phone.fill")
+                }
+            }
+        }
+        .frame(width: 80, height: 40)
+        .background(.ultraThinMaterial)
+        .clipShape(.capsule)
+        .padding()
+        .shadow(radius: 4)
+    }
+
+    private var mailButton: some View {
+        VStack {
+            if viewModel.email == nil {
```

Comment:
Strange code here. You are displaying exactly the same UI right. The only thing is you are make something tappable or not 🤔 Moreover `mailButton` is finally not a button in some cases. And once again `email` not `mail`.
One option could be:
Button { action:... , label: Image() }
.isDisabled(viewModel.email == nil)